graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DataTypeGenerator.kt

```diff
@@ -416,19 +416,36 @@ abstract class BaseDataTypeGenerator(
             getterMethodBuilder.addJavadoc(fieldDefinition.description.sanitizeJavaDoc())
         }
 
-        javaType.addMethod(getterMethodBuilder.build())
-
         val setterName = typeUtils.transformIfDefaultClassMethodExists("set${fieldDefinition.name[0].uppercase()}${fieldDefinition.name.substring(1)}", TypeUtils.Companion.setClass)
-        javaType.addMethod(
-            MethodSpec.methodBuilder(setterName)
-                .addModifiers(Modifier.PUBLIC)
-                .addParameter(returnType, ReservedKeywordSanitizer.sanitize(fieldDefinition.name))
-                .addStatement(
-                    "this.\$N = \$N",
-                    ReservedKeywordSanitizer.sanitize(fieldDefinition.name),
-                    ReservedKeywordSanitizer.sanitize(fieldDefinition.name)
-                ).build()
-        )
+        val parameterBuilder = ParameterSpec.builder(returnType, ReservedKeywordSanitizer.sanitize(fieldDefinition.name))
+        val setterMethodBuilder = MethodSpec.methodBuilder(setterName)
+            .addModifiers(Modifier.PUBLIC)
+            .addStatement(
+                "this.\$N = \$N",
+                ReservedKeywordSanitizer.sanitize(fieldDefinition.name),
+                ReservedKeywordSanitizer.sanitize(fieldDefinition.name)
+            )
+
+        if (fieldDefinition.directives.isNotEmpty()) {
+            val (annotations, comments) = applyDirectives(fieldDefinition.directives)
+            if (!comments.isNullOrBlank()) {
+                fieldBuilder.addJavadoc("\$L", comments)
+            }
+            annotations.forEach { entry ->
```

Comment:
Currently the target sites available:
@get
@set
@file
@param
@field
@setparam
@receiver
@property
@delegate

Targets supported here:
@get
@set
@param
@field
@setparam

Unsupported targets:
**@file** - @file is most commonly used with @file:JvmName in order to provide a custom name for a file. Since the customannotations are mostly on the types, inputs and fields, we did not add support for this.
**@receiver** - In Kotlin a “receiver” is the instance on which an [extension function](https://kotlinlang.org/docs/reference/extensions.html) is defined, or the type for a [lambda with receiver](https://kotlinlang.org/docs/reference/lambdas.html#function-literals-with-receiver). It is essentially the type on which a block of code is intended to run. In the case of a lambda with receiver in Kotlin, the lambda runs as if it is part of the receiving class. In the case of extension functions, the defined function also runs as if it is part of the receiving class type. As this does not apply to types,  inputs or fields I didnt add support for it.
**@property** - @property is the only use-site target that has no effect on the resulting bytecode when viewed from Java. This is due to the fact that Java does not have the same notion of properties as Kotlin. Because of this, adding property-specific annotations will only be useful to kotlin-specific libraries and tools. Since java does not support this I didnt add support for this target in java.
**@delegate** - A delegate is simply a way to wrap common code that will be executed each time you read and/or write from a property value. As this too does not seem to be part of types, inputs and fields, we did not add support for this.

Reply:
@srinivasankavitha I have updated the description with java examples for each of the target sites we support. Please take a look at it and let me know if you have questions.